### DISCUSSION TITLE
Invalid argument(s) 'use_batch_mode' sent to create_engine()

### DISCUSSION THREAD

**Discussion by sameli74** [sqlalchemy/sqlalchemy#6064] [Usage Questions] [answered]
Hello,

I used  `sqlalchemy<1.4.0` with `postgresql+psycopg2` as the flavour and `use_batch_mode= True` in a couple of my codes, and didn't have any problem. However with the new release `1.4.0`, I get the following error:

`Invalid argument(s) 'use_batch_mode' sent to create_engine(), using configuration PGDialect_psycopg2/QueuePool/Engine.  Please check that the keyword arguments are appropriate for this combination of components.`

Can anyone please give me an advice or share with me a link on how to solve this issue.
Thanks.

**Comment by CaselIT** [comment]
Hi,

It's documented in the changelog. Take a loot at [this section](https://docs.sqlalchemy.org/en/14/changelog/migration_14.html?highlight=use_batch_mode#psycopg2-dialect-features-execute-values-with-returning-for-insert-statements-by-default)

**Comment by zzzeek** [comment, marked as answer]
hey there -

the name use_batch_mode was deprecated in 1.3:

https://docs.sqlalchemy.org/en/13/dialects/postgresql.html#psycopg2-executemany-mode


use the parameter name "executemany_mode" instead:

  executemany_mode = "batch"

however this whole area has been improved and you might be better off with just leaving the flag out entirely now - at leasst for INSERTs, you want the default behavior.
